frontend/src/components/Item/MediaStreamSelector.vue

```diff
@@ -1,204 +1,163 @@
 <template>
   <v-select
     v-model="trackIndex"
-    flat
-    dense
+    density="comfortable"
     single-line
     hide-details
     class="text-truncate"
-    :items="selectItems"
-    @input="$emit('input', $event)">
+    :items="selectItems">
     <template #selection="{ item }">
-      {{ item.text.selection }}
+      {{ item.raw.selection }}
     </template>
 
-    <template #item="{ item, on, attrs }">
+    <template #item="{ item, props: templateProps }">
       <v-list-item
-        v-bind="attrs"
-        :lines="!!item.text.subtitle && 'two'"
-        v-on="on">
-        <v-avatar v-if="item.text.icon">
-          <v-icon :icon="item.text.icon" />
-        </v-avatar>
-        <v-list-item-title>{{ item.text.title }}</v-list-item-title>
-        <v-list-item-subtitle v-if="item.text.subtitle">
-          {{ item.text.subtitle }}
-        </v-list-item-subtitle>
-      </v-list-item>
+        v-bind="templateProps"
+        :title="item.raw.title"
+        :subtitle="item.raw.subtitle"
+        :prepend-icon="item.raw.icon" />
     </template>
   </v-select>
 </template>
 
-<script lang="ts">
-import { PropType, defineComponent } from 'vue';
+<script lang="ts" setup>
+import { ref, computed, watch } from 'vue';
 import langs from 'langs';
 import { MediaStream } from '@jellyfin/sdk/lib/generated-client';
+import { useI18n } from 'vue-i18n';
 import IMdiSurroundSound20 from 'virtual:icons/mdi/surround-sound-2-0';
 import IMdiSurroundSound31 from 'virtual:icons/mdi/surround-sound-3-1';
 import IMdiSurroundSound51 from 'virtual:icons/mdi/surround-sound-5-1';
 import IMdiSurroundSound71 from 'virtual:icons/mdi/surround-sound-7-1';
 import IMdiSurroundSound from 'virtual:icons/mdi/surround-sound';
 
-interface SelectItems {
-  selection: string;
-  subtitle: string | undefined;
-  icon: typeof IMdiSurroundSound | undefined;
-  title: string;
-}
+const props = withDefaults(
+  defineProps<{
+    mediaStreams: MediaStream[];
+    type: string;
+    defaultStreamIndex?: number;
+  }>(),
+  { mediaStreams: () => [], defaultStreamIndex: undefined }
+);
+const emits = defineEmits(['input']);
```

Comment:
I believe this is not the most type-safe way to define an emit type. 

Check this for example: https://github.com/jellyfin/jellyfin-vue/blob/vite/frontend/src/components/Layout/Carousel/Carousel.vue#L79

Reply:
True, I missed that it was available in Vue3. Fixed